pkg/pillar/types/persisttypes.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+// PersistType is type of persist we use
+type PersistType uint8
+
+// Enum of PersistType variants
+const (
+	PersistUnknown PersistType = iota
```

Comment:
This corresponds to two different cases; the string is empty (aka unknown) and the string does not match any.
It might make sense to have a separate value for UnInitialized and Unknown aka Bad values. But I don't know if that would make the error reporting more clear.